sync/vm/registry.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package vm
+
+import (
+	"context"
+	"fmt"
+
+	synccommon "github.com/ava-labs/coreth/sync"
+	"github.com/ava-labs/libevm/log"
+)
+
+// SyncerTask represents a single syncer with its name for identification.
+type SyncerTask struct {
+	name   string
+	syncer synccommon.Syncer
+}
+
+// SyncerRegistry manages a collection of syncers for concurrent execution.
+type SyncerRegistry struct {
+	syncers []SyncerTask
+	names   map[string]bool // Track registered names to prevent duplicates.
+}
+
+// NewSyncerRegistry creates a new empty syncer registry.
+func NewSyncerRegistry() *SyncerRegistry {
+	return &SyncerRegistry{
+		syncers: make([]SyncerTask, 0),
+		names:   make(map[string]bool),
+	}
+}
+
+// Register adds a syncer to the registry.
+// Returns an error if a syncer with the same name is already registered.
+func (r *SyncerRegistry) Register(name string, syncer synccommon.Syncer) error {
+	if r.names[name] {
+		return fmt.Errorf("syncer with name '%s' is already registered", name)
+	}
+
+	r.names[name] = true
+	r.syncers = append(r.syncers, SyncerTask{name, syncer})
+
+	return nil
+}
+
+// RunSyncerTasks executes all registered syncers.
+func (r *SyncerRegistry) RunSyncerTasks(ctx context.Context, client *client) error {
+	if len(r.syncers) == 0 {
+		return nil
+	}
+
+	for _, task := range r.syncers {
+		log.Info(task.name+" starting", "summary", client.summary)
```

Comment:
Is it idiomatic to have no space here?
```suggestion
		log.Info(task.name + " starting", "summary", client.summary)
```

Reply:
<img width="688" height="148" alt="Screenshot 2025-07-30 at 0 58 24" src="https://github.com/user-attachments/assets/b7dde36a-cade-4b82-a56b-c4f1d7fc9d85" />
You can see for yourself that `goimports` formatter prefers it without spaces.

Reply:
Awesome, just curious

Reply:
agree on that, will change it